### PR TITLE
Total jobs

### PR DESCRIPTION
**Pull request by bruggerk**
total number of jobs run by galaxy instance split into states

Example output:

```
    state    | count 
-------------+-------
 deleted     |     1
 deleted_new |     1
 error       |    10
 ok          |    21
```

### CODE REVIEW

File: parts/22-query.sh [hunk 1]
```diff
@@ -1692,6 +1692,33 @@ query_monthly-jobs(){ ## [year]: Number of jobs run each month
 
 }
 
+query_total-jobs(){ ## [year]: Total number of jobs run by galaxy instance
+	handle_help "$@" <<-EOF
+		Count total number of jobs
+
+		    [galaxy@sn04 galaxy]$ gxadmin query monthly-jobs 2018
```
Comment (hexylena) [inline parts/22-query.sh:1699]:
```suggestion
		    $ gxadmin query jobs-total
```

File: parts/22-query.sh [hunk 2]
```diff
@@ -1692,6 +1692,33 @@ query_monthly-jobs(){ ## [year]: Number of jobs run each month
 
 }
 
+query_total-jobs(){ ## [year]: Total number of jobs run by galaxy instance
```
Comment (hexylena) [inline parts/22-query.sh:1695]:
```suggestion
query_jobs-total(){ ## : Total number of jobs by state
```

Comment (hexylena) [reply parts/22-query.sh:1695]: The `[year]` here indicates an optional parameter and I think none is required here.

File: parts/22-query.sh [hunk 3]
```diff
@@ -1692,6 +1692,33 @@ query_monthly-jobs(){ ## [year]: Number of jobs run each month
...
+		     ok          |    21
+	EOF
+
+
+	read -r -d '' QUERY <<-EOF
+		SELECT
+			state, count(*)
+		FROM
+			job
+                GROUP BY
```
Comment (hexylena) [inline parts/22-query.sh:1714]: Could you please use tabs here?

**Comment by hexylena** [comment]
You might also be interested in the internal, hidden function:

```
$ gxadmin query server-jobs
```

the output of which includes the job runner + destination name.

**Comment by bruggerk** [comment]
Hi

I have done the changes you asked for and it now passes the build process. I have furthermore fixed it so it can now do the iquery as well. 

Please let me know if there is anything you need, or if something fails

Thanks 

Kim

File: docs/README.query.md [hunk 4]
```diff
@@ -1532,10 +1555,28 @@ query ts-repos -  Counts of toolshed repositories by toolshed and owner.
...
+
+    gxadmin query upload-gb-in-past-hour [hours|1]
+gxadmin query upload-gb-in-past-hour [hours|1]
+
+**NOTES**
+
+Quick output, mostly useful for graphing, to produce a nice graph of how heavily are people uploading currently.
+
+
+## query upload-gb-in-past-hour
```
Comment (bgruening) [inline docs/README.query.md:1571]: this is redundant as it seems...

File: parts/22-query.sh (hunk 1 above)
Comment (bgruening) [reply parts/22-query.sh:1699]: ping

File: docs/README.query.md (hunk 4 above)
Comment (hexylena) [reply docs/README.query.md:1571]: I'll look into it.

File: parts/22-query.sh [hunk 5]
```diff
@@ -1692,6 +1692,37 @@ query_monthly-jobs(){ ## [year]: Number of jobs run each month
 
 }
 
+query_total-jobs(){ ## [year]: Total number of jobs run by galaxy instance
+	handle_help "$@" <<-EOF
+		Count total number of jobs
+
+		    [galaxy@sn04 galaxy]$ gxadmin query monthly-jobs 2018
```
Comment (bgruening) [inline parts/22-query.sh:1699]: this is still wrong ...

Comment (hexylena) [reply parts/22-query.sh:1699]: I'm cleaning them all, I'll get this while I'm there.

File: docs/README.query.md (hunk 4 above)
Comment (hexylena) [reply docs/README.query.md:1571]: It seems something went wrong on their end, compiling the docs. I've re-generated them locally and looks perfect so ??

**Pull request merged**